codegen-server/src/main/kotlin/software/amazon/smithy/rust/codegen/server/smithy/generators/ServerServiceGeneratorV2.kt

```diff
@@ -469,8 +472,108 @@ class ServerServiceGeneratorV2(
     }
 
     fun render(writer: RustWriter) {
+        val crateName = codegenContext.moduleUseName()
+        val handlers: Writable = operations
+            .map { operation ->
+                DocHandlerGenerator(operation, "///", builderFieldNames[operation]!!, codegenContext).docSignature()
+            }
+            .reduce { acc, wt ->
+                writable {
+                    rustTemplate("#{acc:W} \n#{wt:W}", "acc" to acc, "wt" to wt)
+                }
+            }
+
+        val hasErrors = service.operations.any { model.expectShape(it).asOperationShape().get().errors.isNotEmpty() }
+
         writer.rustTemplate(
             """
+            /// A fast and customizable Rust implementation of the $serviceName Smithy service.
+            ///
+            /// The primary export is [`$serviceName`]: it satisfies the [`Service<http::Request, Response = http::Response>`]
+            /// trait and therefore can be handed to [Hyper server] using [`$serviceName::into_make_service`] or used in Lambda using [`#{SmithyHttpServer}::routing::LambdaHandler`].
```

Comment:
This is nice. We _could_ have a doc test here showing it being passed into `hyper`.

```rust
# let server = async { /* some fake Service which makes the below compile OR construct the entire service for better coverage */ };
let server = app.into_make_service();
let server = hyper::Server::bind(&bind).serve(server);
```

```rust
/* same thing for lambda */
```